build.gradle.kts

```diff
@@ -106,54 +101,57 @@ subprojects {
 
     // ==== Tests ====
     // https://docs.gradle.org/current/samples/sample_java_multi_project_with_junit5_tests.html
-    test {
+    tasks.test {
         useJUnitPlatform()
     }
+
     // Log on passed, skipped, and failed test events if the `-Plog-tests` property is set.
     // https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/testing/logging/TestLoggingContainer.html
     if (project.hasProperty("log-tests")) {
-        test {
+        tasks.test {
             testLogging {
-                events = ["passed", "skipped", "failed"]
-                exceptionFormat = "full"
+                events("passed", "skipped", "failed")
+                exceptionFormat = TestExceptionFormat.FULL
             }
         }
     }
 
     // ==== Maven ====
-    apply plugin: "maven-publish"
-    apply plugin: "signing"
-    apply plugin: "com.github.johnrengelman.shadow"
+    apply(plugin = "maven-publish")
+    apply(plugin = "signing")
+    apply(plugin = "com.github.johnrengelman.shadow")
 
     // This is a little hacky, but currently needed to build a shadowed CLI JAR and smithy-syntax JAR with the same
     // customizations as other JARs.
     if (project.name != "smithy-cli" && project.name != "smithy-syntax") {
-        tasks.shadowJar.enabled = false
+        tasks.named<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar>("shadowJar") {
+            isEnabled = false
+        }
     }
 
     publishing {
         repositories {
             // JReleaser's `publish` task publishes to all repositories, so only configure one.
             maven {
                 name = "localStaging"
-                url = stagingDirectory
+                url = uri(stagingDirectory)
             }
         }
 
         publications {
-            mavenJava(MavenPublication) { publication ->
-                if (tasks.shadowJar.enabled) {
-                    project.shadow.component(publication)
+            create<MavenPublication>("mavenJava") {
+                if (tasks.findByName("shadowJar")?.enabled == true) {
```

Comment:
```suggestion
                if (tasks.findByName("shadowJar")?.enabled) {
```

Reply:
Hah Classic mistake!! :D I never thought I would make this but in these many files in hindsight I did. Will update.